master/docs/examples/gitlab.cfg

```diff
@@ -0,0 +1,223 @@
+c['workers'] = [worker.Worker("buildbot-worker", "buildbot-pass")]
+workernames = ["buildbot-worker"]
+
+# 'protocols' contains information about protocols which master will use for
+# communicating with workers. You must define at least 'port' option that workers
+# could connect to your master with this protocol.
+# 'port' must match the value configured into the workers (with their
+# --master option)
+c['protocols'] = {'pb': {'port': 9989}}
+
+####### CHANGESOURCES
+
+# the 'change_source' setting tells the buildmaster how it should find out
+# about source code changes.
+
+c['change_source'] = []
+
+####### SCHEDULERS AND BUILDERS
+
+# The Schedulers decide how to react to incoming changes.  
+c['schedulers'] = []
+# The 'builders' list defines the Builders, which tell Buildbot how to
+# perform a build: what steps, and which workers can execute them.
+# Note that any particular build will only take place on one worker.
+c['builders'] = []
+
+# Call addGitLabProject once for each similar project you want to build.
+# It adds factories, schedulers, and builders for the project
+# in one fell swoop, making it easier to have a large number
+# of similar projects.
+addGitLabProject('git@gitlab.example.com:build/gnu-hello', workernames)
+addGitLabProject('git@gitlab.example.com:build/gnu-time', workernames)
+
+####### BUILDBOT SERVICES
+
+# 'services' is a list of BuildbotService items like reporter targets. The
+# status of each build will be pushed to these targets. buildbot/reporters/*.py
+# has a variety to choose from, like IRC bots.
+
+c['services'] = []
+
+####### PROJECT IDENTITY
+
+# the 'title' string will appear at the top of this buildbot installation's
+# home pages (linked to the 'titleURL').
+
+c['title'] = "Gnu Hello GitLab"
+c['titleURL'] = "https://gitlab.example.com/build/"
+
+# the 'buildbotURL' string should point to the location where the buildbot's
+# internal web server is visible. This typically uses the port number set in
+# the 'www' entry below, but with an externally-visible host name which the
+# buildbot cannot figure out without some help.
+
+c['buildbotURL'] = "http://buildbot.example.com:8010/"
+
+# CONFIGME: buildbot authentication
+# This example tries to show nothing to anonymous users.
+# authz = util.Authz(
+#   allowRules=[
+#     util.AnyEndpointMatcher(role="platform"),
+#     util.AnyEndpointMatcher(role="snorglepuss-does-not-exist", defaultDeny=True),
+#   ],
+#   roleMatchers=[
+#     util.RolesFromGroups()
+#   ]
+# )
+
+# minimalistic config to activate new web UI
+c['www'] = dict(
+    port=8010,
+    # CONFIGME: GitLab application
+    # Enter the URL of your buildbot gitlab hook into the GitLab UI at
+    #  "User Settings / Applications / Add New Application"
+    # to obtain the 2nd and 3rd parameters for GitLabAuth.
+    auth=util.GitLabAuth("https://gitlab.example.com",
+                         "036777777777777777777777777777777777777777777777777777777cea803a",
```

Comment:
probably you should rather put <userid> and <userpasswd> instead of those weird long strings

Reply:
ok.  Those are generated by gitlab, I think, and the
strings I gave are the actual length of the hex strings generated
by gitlab.  But I can use the names gitlab uses intead of random examples with right length.